### ISSUE TITLE
support pathlib.Path objects as filepath?

### ISSUE THREAD

**Issue by raphaelquast** [GeospatialPython/pyshp#234]
Hey, I've just noticed that the reader can't deal with `pathlib.Path` objects... ([standard python3 - pathlib](https://docs.python.org/3/library/pathlib.html))

I guess it should be an easy fix (a simple `str()` for any path-like object already does the trick...) 
```python
from pathlib import Path
import shapefile
shppath = Path(".... path to the shapefile")

r = shapefile.Reader(shppath)         # this will fail
r.records()

r = shapefile.Reader(str(shppath))    # works as expected
r.records()
```

**Comment by karimbahgat** [comment]
Thanks for putting this on the radar. Support for pathlib and other path-like objects will be included in next version, see #233.

**Comment by raphaelquast** [comment]
awesome! nice to hear :-)
